### PR TITLE
Fix warning in logs while moving FutureFile after chunk assembly

### LINKED ISSUE #31631: Moving FutureFile after chunk assembly logs warning
### Steps

1. ~~Run acceptance test from https://github.com/owncloud/guests/pull/212 with name features/apiGuests/guests.feature:44~~
1. ~`cd tests/acceptance; ./run.sh features/apiWebdav/webdav-related-new-endpoint.feature:497`~
EDIT vicdeo:
since https://github.com/owncloud/core/commit/a55ce776c336831cc4e0c258c5780a81bd5a64c1#diff-4ab58a91bb1ef5af35805d89e6475100 it's
`cd tests/acceptance; ./run.sh ./run.sh features/apiWebdav/webdav-related-new-endpoint.feature:38`

1. Observe output then search for ".file"

### Expected
No warning

### Actual
Warning appears:
```
{"reqId":"nPvqv4sBxY2z8A77udaO","level":2,"time":"2018-06-04T13:04:12+00:00","remoteAddr":"127.0.0.1","user":"guest@example.com","app":"dav","method":"MOVE","url":"\/remote.php\/dav\/uploads\/guest@example.com\/chunking-42\/.file","message":"Could not get node for path: \"uploads\/guest@example.com\/chunking-42\/.file\" : File with name \/\/chunking-42 could not be located"}
```

### Version
With stable10 51142893f67331533d4b94b8db1bf9bb92d61c3d and guests PR which adds a new test https://github.com/owncloud/guests/pull/212.

### TODO
- [x] debug for stack trace
```
0  OCA\DAV\DAV\MiscCustomPropertiesBackend->delete() /srv/www/htdocs/owncloud/apps/dav/lib/DAV/MiscCustomPropertiesBackend.php:55
1  Sabre\DAV\PropertyStorage\Plugin->afterUnbind() /srv/www/htdocs/owncloud/lib/composer/sabre/dav/lib/DAV/PropertyStorage/Plugin.php:126
2  call_user_func_array:{/srv/www/htdocs/owncloud/lib/composer/sabre/event/lib/EventEmitterTrait.php:105}() /srv/www/htdocs/owncloud/lib/composer/sabre/event/lib/EventEmitterTrait.php:105
3  OCA\DAV\Connector\Sabre\Server->emit() /srv/www/htdocs/owncloud/lib/composer/sabre/event/lib/EventEmitterTrait.php:105
4  Sabre\DAV\CorePlugin->httpMove() /srv/www/htdocs/owncloud/lib/composer/sabre/dav/lib/DAV/CorePlugin.php:649
5  call_user_func_array:{/srv/www/htdocs/owncloud/lib/composer/sabre/event/lib/EventEmitterTrait.php:105}() /srv/www/htdocs/owncloud/lib/composer/sabre/event/lib/EventEmitterTrait.php:105
6  OCA\DAV\Connector\Sabre\Server->emit() /srv/www/htdocs/owncloud/lib/composer/sabre/event/lib/EventEmitterTrait.php:105
7  OCA\DAV\Connector\Sabre\Server->invokeMethod() /srv/www/htdocs/owncloud/lib/composer/sabre/dav/lib/DAV/Server.php:479
8  OCA\DAV\Connector\Sabre\Server->exec() /srv/www/htdocs/owncloud/lib/composer/sabre/dav/lib/DAV/Server.php:254
9  OCA\DAV\Server->exec() /srv/www/htdocs/owncloud/apps/dav/lib/Server.php:279
10 require_once()  /srv/www/htdocs/owncloud/apps/dav/appinfo/v2/remote.php:31
11 {main}          /srv/www/htdocs/owncloud/remote.php:165
```

- [x] confirm whether this also happens outside of guest app
- [x] check if regression
- [ ] find good solution

@VicDeo I believed you touched this logic recently as part of https://github.com/owncloud/core/pull/31093 and might understand it better.
The code states that this is supposed to be a "rare" situation here: https://github.com/owncloud/core/blob/stable10/apps/dav/lib/DAV/AbstractCustomPropertiesBackend.php#L245
But the reason the node is not found is because it was already moved (actually copied + deleted) after chunk assembly.

Would it be safe to ignore the exception here and prevent the log entry or is there some bigger issue hidden in there ?
Should this code also be covered by some logic grabbing the nodes beforehand like in https://github.com/owncloud/core/pull/31093 ?

### PR DESCRIPTION
**Pull request by VicDeo**
## Description
Register FileCustomPropertiesBackend for `files/`  and `uploads/` 
 MiscCustomPropertiesBackend for everything else

## Related Issue
Fixes https://github.com/owncloud/core/issues/31631

## Motivation and Context
Extra log entry

## How Has This Been Tested?
1. By uploading big files
also 
2. 
`cd tests/acceptance; ./run.sh ./run.sh features/apiWebdav/webdav-related-new-endpoint.feature:38`
`grep 'Could not get node for path' ../../data/owncloud.log` 
#### expected:
nothing
#### actual
```
{"reqId":"YsWCEJL9CwgkRKFVOAnr","level":2,"time":"2018-07-04T14:24:39+00:00","remoteAddr":"127.0.0.1","user":"user0","app":"dav","method":"MOVE","url":"\/remote.php\/dav\/uploads\/user0\/chunking-42\/.file","message":"Could not get node for path: \"uploads\/user0\/chunking-42\/.file\" : File with name \/\/chunking-42 could not be located"}
```

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by PVince81 — COMMENTED** [review]
Change looks fine. Any chance to make unit tests happy ?

**Comment by VicDeo** [comment]
@PVince81 testing it is tricky. This class is a pure constructor and the changed lines are wrapped with 
```
$this->server->on('beforeMethod', function () use ($root) {
```

**Comment by PVince81** [comment]
ok, let's do it with acceptance test only, that should be enough

**Comment by phil-davis** [comment]
We need a codecov that can measure code exercised by acceptance tests, and do a union of acceptance and unit test coverage.

**Comment by PVince81** [comment]
@phil-davis can you guide @VicDeo for the acceptance tests part ?

**Comment by phil-davis** [comment]
@individual-it please have a quick look. If not easy, then assign me so I can help next week.

**Comment by individual-it** [comment]
added acceptance tests @phil-davis please review those

File: tests/acceptance/features/bootstrap/LoggingContext.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ */
+
+use Behat\Behat\Context\Context;
+use Behat\Behat\Hook\Scope\BeforeScenarioScope;
+use TestHelpers\SetupHelper;
+
+require_once 'bootstrap.php';
+
+/**
+ * Defines test steps specific to the antivirus app
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/LoggingContext.php]: antivirus??? @individual-it

**Review by phil-davis — COMMENTED** [review]
Acceptance test changes are OK

**Comment by VicDeo** [comment]
@individual-it 
Great job, thank you!

**Comment by VicDeo** [comment]
@PVince81 acceptance tests added, someone with merge powers is needed.

**Comment by PVince81** [comment]
restarted failed drone build (well, it says "cancelled" despite phpunit having finished) for more greenery

**Comment by PVince81** [comment]
@VicDeo please backport

**Comment by VicDeo** [comment]
Stable10: https://github.com/owncloud/core/pull/32166

**Pull request merged**